src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/BufferBlockCopyLengthTests.cs

```diff
@@ -0,0 +1,453 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.BufferBlockCopyLengthAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.BufferBlockCopyLengthAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class BufferBlockCopyLengthTests
+    {
+        [Fact]
+        public async Task SrcIsByteArray()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        byte[] src = new byte[] {1, 2, 3, 4};
+        byte[] dst = new byte[] {0, 0, 0, 0};
+        
+        Buffer.BlockCopy(src, 0, dst, 0, src.Length);
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Byte() {1, 2, 3, 4}
+        Dim dst = New Byte() {0, 0, 0, 0}
+
+        Buffer.BlockCopy(src, 0, dst, 0, src.Length)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task DstIsByteArray()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        byte[] src = new byte[] {1, 2, 3, 4};
+        byte[] dst = new byte[] {0, 0, 0, 0};
+        
+        Buffer.BlockCopy(src, 0, dst, 0, dst.Length);
+        
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Byte() {1, 2, 3, 4}
+        Dim dst = New Byte() {0, 0, 0, 0}
+
+        Buffer.BlockCopy(src, 0, dst, 0, dst.Length)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
```

Comment:
1 cool thing you can do here is to use a Theory instead of a `Fact`. With Theory you can write these 4 test case (byte, sbyte, src.Length, dst.Length) with just 1 function. See `TestAppendLiteral` in `PreferConstCharOverConstUnitStringTests.cs` to get an idea of how it works and https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/#:~:text=xUnit%20uses%20the%20%5BFact%5D%20attribute%20to%20denote%20a,the%20most%20common%20is%20with%20an%20%5BInlineData%5D%20attribute. for an overview of `Theory` vs `Fact`